Added support for sprite Vector array and Static array in center_camera_on() method

MangoS9 opened the pull request:
# Description
Added in support for vector array and static array for center_camera_on() method for C++ as calculating multiple sprite locations can be tedious.
```cpp
void center_camera_on(vector<sprite> s, double offset_x, double offset_y);
```
parameters:

- `vector<sprite> s`
- `double offset_x`
- `double offset_y`

```cpp
void center_camera_on(sprite s[],int size , double offset_x, double offset_y);
```
parameters:

- `sprite s[]`
- `int size`
- `double offset_x`
- `double offset_y`
Both supports vector_2d data sets too:
```cpp
	//static arr
	void center_camera_on(sprite s[],int size , const vector_2d &offset);
	//vector arr
	void center_camera_on(vector<sprite> s, const vector_2d &offset)
```

**Video demo:**

https://github.com/splashkit/splashkit-core/assets/128771372/ab3c07b4-edf8-41b1-b600-25aa9ed87094


## Type of change

- [x] Overloading the method in camera.cpp and camera.h file
- [x] Modified the camera test to check the functionality 

# How Has This Been Tested?

- [x] Tested with Windows 10
- [x] Test on single variable, vector array and static array
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Requested a review from seniors on the Pull Request

macite commented on coresdk/src/coresdk/camera.cpp:
```diff
@@ -168,12 +168,60 @@ namespace splashkit_lib
-    void center_camera_on(sprite s, double offset_x, double offset_y)
-    {
-        point_2d center = sprite_position(s);
+    // void center_camera_on(sprite s, double offset_x, double offset_y)
```
These comments should not remain in the code. Only include comments that add value.

macite commented on coresdk/src/coresdk/camera.cpp:
```diff
@@ -168,12 +168,60 @@ namespace splashkit_lib
+		double sc_x{0};
+		double sc_y{0};
+		
+        for (const auto& ele : s) {
```
don't use auto

macite commented on coresdk/src/coresdk/camera.h:
```diff
@@ -330,6 +330,50 @@ namespace splashkit_lib
+	 * @param offset_y An additional offset added to the camera, allowing you to
+	 *                 position the sprites offset from the center of the screen.
+	 */
+	void center_camera_on(sprite s[],int size , double offset_x, double offset_y);
```
I don't think the translator supports arrays - have you tested rebuilding the language translations. See the `tools/scripts/deploy.sh` script.

macite commented on coresdk/src/coresdk/camera.h:
```diff
@@ -330,6 +330,50 @@ namespace splashkit_lib
+	 * @param offset   The amount to offset the camera, allowing you to position
+	 *                 the sprites away from the center of the screen.
+	 */
+	void center_camera_on(sprite s[], int size, const vector_2d &offset);
```
As above

macite commented on coresdk/src/coresdk/camera.h:
```diff
@@ -330,6 +330,50 @@ namespace splashkit_lib
      * @attribute method center_on
      */
     void center_camera_on(sprite s, const vector_2d &offset);
+	/**
```
Very sloppy!

Match indentation.

macite commented on coresdk/src/test/test_camera.cpp:
```diff
@@ -19,39 +20,67 @@ void run_camera_test()
 
     load_bitmap("ufo", "ufo.png");
-    sprite s = create_sprite("ufo");
+    sprite s1 = create_sprite("ufo");
```
Why not just in the vector and then access them there by index when needed.

macite replied:
Then use a loop to run repeated instructions

macite commented on coresdk/src/test/test_camera.cpp:
```diff
@@ -19,39 +20,67 @@ void run_camera_test()
         }
 
-        update_sprite(s);
+        update_sprite(s1);
```
Loop

macite commented on coresdk/src/test/test_camera.cpp:
```diff
@@ -65,7 +94,9 @@ void run_camera_test()
         draw_triangle(COLOR_AQUA, screen_width() / 2, 0, 0, screen_height(), screen_width(), screen_height());
 
-        draw_sprite(s);
+        draw_sprite(s1);
```
Loop

macite commented on coresdk/src/coresdk/camera.cpp:
```diff
@@ -182,4 +230,14 @@ namespace splashkit_lib
         center_camera_on(s, offset.x, offset.y);
     }
+	
+	void center_camera_on(sprite s[],int size , const vector_2d &offset)
```
Sloppy!

macite reviewed (changes requested):
This needs work, and should have been reviewed before submitting here. Reviews should critically check basic things like indentation.

MangoS9 commented:
I will make the changes once I got back to my computer 🙌

MangoS9 commented:
strangely enough, I didnt notice the indentation error when I'm working on it. Usually VS code should have warned me about it 🙁

MangoS9 commented:
Update: It seems like the indentation on my local computer looks fine, I do not know why this happens in github but I will get it fixed real soon and have my senior to get it checked up

MangoS9 commented:
Update: 
- fixed the indentation (Hopefully)
- Reformat the test_camera code into a more readable manner
- Changed "auto" type to "sprite"
- Removed all the unnecessary comment in the camera.h
